filebeat/publish/publish.go

```diff
@@ -172,9 +172,12 @@ func (p *asyncLogPublisher) Start() {
 			case <-p.done:
 				return
 			case events := <-p.in:
-				pubEvents := make([]common.MapStr, len(events))
-				for i, event := range events {
-					pubEvents[i] = event.ToMapStr()
+
+				pubEvents := make([]common.MapStr, 0)
```

Comment:
Changed. For the cleanup: I plan a second PR after this one that cleans up the publisher. But I didn't do it in this one as this one should be backported and I want to keep the changes as minimal as possible.